noir-projects/noir-contracts/contracts/card_game_contract/src/cards.nr

```diff
@@ -120,42 +121,46 @@ impl Deck<&mut PrivateContext> {
         inserted_cards
     }
 
-    pub fn get_cards<N>(&mut self, cards: [Card; N]) -> [CardNote; N] {
+    pub fn get_cards_and_note_hashes<N>(&mut self, cards: [Card; N]) -> ([CardNote; N], [Field; N]) {
```

Comment:
This function became a bit ugly with the notes and hashes because it didn't work well with the custom `CardNote` struct. But I decided against doing a larger refactor since the ugliness is mostly just in this contract.